### PR TITLE
fix!: extract MP3 encoder plugin

### PR DESCRIPTION
**Pull request by MartinCupela**
### 🎯 Goal

Make lamejs a peer dependency so that the source code is not bundled within the integrator application source code. This is to prevent issues when integrators do not want to bundle code issued under LGPL license.

### 🛠 Implementation details

Emojis and mp3 encoding function have been moved to separate `plugins` folder. However, the emojis import path stays the same.

A new audio recording configuration parameter has been introduced - `encoder` - to allow integrators to plug in the MP3 or other encoder. 

### 🎨 UI Changes

None

BREAKING CHANGE: MP3 audio encoder has to be explicitly imported and used as a plugin for audio recordings. The default audio recording format is audio/wav.
BREAKING CHANGE: @breezystack/lamejs became a peer dependency and has to be installed by the integrator so that the MP3 audio encoder can work properly.

### CODE REVIEW

File: docusaurus/docs/React/components/message-input-components/audio-recorder.mdx [hunk 1]
```diff
@@ -51,6 +51,21 @@ The dialog can be customized by passing own component to `Channel` component con
 <Channel RecordingPermissionDeniedNotification={CustomComponent}>
 ```
 
+## Custom encoding
+
+By default, the recording is encoded into `audio/wav` format. In order to reduce the size and keep the inter-browser format compatibility, you can use an MP3 encoder that is based on [`lamejs` implementation](https://github.com/gideonstele/lamejs). Follow these steps to achieve the MP3 encoding capability.
+
+1. The library `@breezystack/lamejs` has to be installed as this is a peer dependency to `stream-chat-react`.
```
Comment (oliverlaz) [inline docusaurus/docs/React/components/message-input-components/audio-recorder.mdx:58]: I think it would be good if we add an install command here. Easy to copy-paste, instead of having to type the full command manually.

File: docusaurus/docs/React/release-guides/upgrade-to-v12.mdx [hunk 2]
```diff
@@ -4,6 +4,40 @@ title: Upgrade to v12
...
 ---
 
+## Audio recordings transcoding
+
+Until now, the audio recordings were transcoded to `audio/mp3` format for inter-browser compatibility and size reduction. However, as of the v12, the MIME type `audio/wav` will be the default. The mp3 encoder used by the library is licensed under LGPL and integrators will have to opt-in to using the library in their apps.
+
+:::important
+**Action required**<br/>
+
+1. The library `@breezystack/lamejs` has to be installed as this is a peer dependency to `stream-chat-react`.
```
Comment (oliverlaz) [inline docusaurus/docs/React/release-guides/upgrade-to-v12.mdx:14]: Install command here as well.

File: docusaurus/docs/React/release-guides/upgrade-to-v12.mdx [hunk 3]
```diff
@@ -4,6 +4,40 @@ title: Upgrade to v12
 keywords: [migration guide, upgrade, v12, breaking changes]
 ---
 
+## Audio recordings transcoding
+
+Until now, the audio recordings were transcoded to `audio/mp3` format for inter-browser compatibility and size reduction. However, as of the v12, the MIME type `audio/wav` will be the default. The mp3 encoder used by the library is licensed under LGPL and integrators will have to opt-in to using the library in their apps.
```
Comment (oliverlaz) [inline docusaurus/docs/React/release-guides/upgrade-to-v12.mdx:9]: > The mp3 encoder used by the library is licensed under LGPL and integrators will have to opt-in to using the library in their apps.

This sentence creates uncertainty/fear/doubt. I think we should rephrase this to something else, or remove it. IMO, we are compliant with LGPL now. We make mp3 encoding opt-in.

File: docusaurus/react-docusaurus-dontent-docs.plugin.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ module.exports = {
         lastVersion: 'current',
```
Comment (oliverlaz) [inline docusaurus/react-docusaurus-dontent-docs.plugin.js:6]: Until we release v12-final, I think this should point to `11.x.x`

File: src/plugins/Emojis/EmojiPicker.tsx [hunk 5]
```diff
@@ -6,7 +6,7 @@ import Picker from '@emoji-mart/react';
 import type { Options } from '@popperjs/core';
 
 import { useMessageInputContext, useTranslationContext } from '../../context';
-import { EmojiPickerIcon } from '../MessageInput/icons';
+import { EmojiPickerIcon } from './icons';
```
Comment (arnautov-anton) [inline src/plugins/Emojis/EmojiPicker.tsx:9]: 👍

File: docusaurus/react-docusaurus-dontent-docs.plugin.js (hunk 4 above)
Comment (MartinCupela) [reply docusaurus/react-docusaurus-dontent-docs.plugin.js:6]: But then we will have docs paths with v12 in them. Is that ok?

Comment (oliverlaz) [reply docusaurus/react-docusaurus-dontent-docs.plugin.js:6]: yes, this is fine.

Comment (MartinCupela) [reply docusaurus/react-docusaurus-dontent-docs.plugin.js]: Updated

File: docusaurus/docs/React/release-guides/upgrade-to-v12.mdx (hunk 3 above)
Comment (MartinCupela) [reply docusaurus/docs/React/release-guides/upgrade-to-v12.mdx]: Updated

File: docusaurus/docs/React/components/message-input-components/audio-recorder.mdx (hunk 1 above)
Comment (MartinCupela) [reply docusaurus/docs/React/components/message-input-components/audio-recorder.mdx:58]: updated

**Review by oliverlaz — APPROVED** [review]
